bbq/vm/linker.go

```diff
@@ -176,24 +100,107 @@ func LinkGlobals(
 			globals[i] = contractVariable
 			indexedGlobals.Set(contract.Name, contractVariable)
 		case *bbq.ImportedGlobal:
-			// ignore imported globals, they are already linked and will be added later
-			continue
+			importedGlobal := linkImportedGlobal(
+				memoryGauge,
+				location,
+				typedGlobal,
+				context,
+				linkedGlobalsCache,
+			)
+			globals[i] = importedGlobal
+
+			// Don't need to add to the `indexedGlobals`, since, like the below comment says,
+			// importer/caller doesn't need to know globals of nested imports
 		}
 	}
 
-	// add imported globals to the end of the globals
-	globals = append(globals, importedGlobals...)
```

Comment:
This was still causing to have some implicit dependency of order, between the compiler and VM. e.g: As mentioned in the PR description imports **were** linked through `bbq.Program.Import`, but needed them to be in the same order as `bbq.Program.Globals`, because the indexes uses the later.

Instead, directly link using the `bbq.Program.Global` (above), now that it has sufficient information.